java.io.NotSerializableException on  play.api.libs.json.ImmutableLinkedHashMap

We have a redis cache implementation on our scala project to perform caching of some objects.
It has an api similar to PlayCache api and it worked with play-json 2.9.4. 
Internally we use the java serialization (via ObjectInputStream/ObjetOutputStream)  during call to redis set/get operations.
We are migrating to latest Play 2.9.0 version and the play-json 2.10.2 . And after migration during serialization we got following exception if serialized object contains JsObject as it's member  (or child member contains JsObject...)
Can you mark play.api.libs.json.ImmutableLinkedHashMap as serializable ?

```
Caused by: java.io.NotSerializableException: play.api.libs.json.ImmutableLinkedHashMap
at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1187)
at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1572)
at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1438)
at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1181)
at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1572)
at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1438)
at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1181)
at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1572)
at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1529)
at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1438)
at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1181)
at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:350)
at io.waylay.redis.RedisCache$.anySerialize(RedisCache.scala:43)
at io.waylay.redis.RedisCache$.io$waylay$redis$RedisCache$$$anonfun$vSerializer$1(RedisCache.scala:26)
at io.waylay.redis.RedisCache$$anonfun$2.serialize(RedisCache.scala:26)
at redis.api.strings.Set.<init>(Strings.scala:125)
at redis.commands.Strings.set(Strings.scala:81)
at redis.commands.Strings.set$(Strings.scala:76)
at redis.RedisClient.set(Redis.scala:78)
at io.waylay.redis.RedisCache.set(RedisCache.scala:64)
```

--- a/play-json/jvm/src/test/scala/play/api/libs/json/JsObjectSerializationSpec.scala
+++ b/play-json/jvm/src/test/scala/play/api/libs/json/JsObjectSerializationSpec.scala
@@ -1,0 +1,40 @@
+/*
+ * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+import org.specs2.mutable._
+import play.api.libs.json.Json._
+
+import java.io.{ ByteArrayInputStream, ByteArrayOutputStream, ObjectInputStream, ObjectOutputStream }
+
+class JsObjectSerializationSpec extends Specification {
+  "JsObject" should {
+    "serialize/deserialize correctly" in {
+      val originalObj = Json.obj(
+        "field1" -> 123,
+        "field2" -> "abc",
+        "field3" -> JsNull,
+        "obj"    -> Json.obj("field1" -> 234),
+        "arr" -> JsArray(
+          Seq(
+            JsString("abc"),
+            JsNumber(123),
+            JsBoolean(true),
+            JsNull,
+            Json.obj("field1" -> 345)
+          )
+        )
+      )
+
+      val bos = new ByteArrayOutputStream()
+      val out = new ObjectOutputStream(bos)
+      out.writeObject(originalObj)
+
+      val bis = new ByteArrayInputStream(bos.toByteArray)
+      val in  = new ObjectInputStream(bis)
+      in.readObject().asInstanceOf[JsObject].must(beEqualTo(originalObj))
+    }
+  }
+}

--- a/play-json/shared/src/main/scala-2.13+/play/api/libs/json/ImmutableLinkedHashMap.scala
+++ b/play-json/shared/src/main/scala-2.13+/play/api/libs/json/ImmutableLinkedHashMap.scala
@@ -13,7 +13,10 @@ import scala.collection.mutable
 /**
  * Wraps a Java LinkedHashMap as a Scala immutable.Map.
  */
-private[json] class ImmutableLinkedHashMap[A, +B](underlying: JLinkedHashMap[A, B]) extends AbstractMap[A, B] {
+@SerialVersionUID(-2338626292552177485L)
+private[json] class ImmutableLinkedHashMap[A, +B](underlying: JLinkedHashMap[A, B])
+    extends AbstractMap[A, B]
+    with Serializable {
 
   override def get(key: A): Option[B] = Option(underlying.get(key))
 

--- a/play-json/shared/src/main/scala-2.13-/play/api/libs/json/ImmutableLinkedHashMap.scala
+++ b/play-json/shared/src/main/scala-2.13-/play/api/libs/json/ImmutableLinkedHashMap.scala
@@ -17,10 +17,11 @@ import scala.collection.mutable.ArrayBuffer
 /**
  * Wraps a Java LinkedHashMap as a Scala immutable.Map.
  */
+@SerialVersionUID(-2338626292552177485L)
 private[json] class ImmutableLinkedHashMap[A, +B](underlying: JLinkedHashMap[A, B])
     extends AbstractMap[A, B]
     with Map[A, B]
-    with MapLike[A, B, ImmutableLinkedHashMap[A, B]] {
+    with MapLike[A, B, ImmutableLinkedHashMap[A, B]] with Serializable {
 
   override def get(key: A): Option[B] = Option(underlying.get(key))
 

--- a/play-json/shared/src/main/scala/play/api/libs/json/JsValue.scala
+++ b/play-json/shared/src/main/scala/play/api/libs/json/JsValue.scala
@@ -33,6 +33,7 @@ object JsValue {
  * Represents a Json null value.
  */
 case object JsNull extends JsValue {
+  @transient
   implicit val reads: Reads[JsNull.type] = Reads[JsNull.type] {
     case JsNull => JsSuccess(JsNull)
     case _      => JsError("error.expected.null")

